plugin/utils/compiler_builtins.py

```diff
@@ -0,0 +1,197 @@
+        the compiler later for its default flags.
+        """
+        if compiler.endswith("++"):
+            return "c++"
+        else:
+            return "c"
+
+    def _get_default_flags(self, compiler, std, language):
+        import subprocess
+        import re
+
+        result = list()
+
+        args = [compiler, "-x", language]
+        if std is not None:
+            args += ['-std=' + std]
+        args += ["-dM", "-E", "-"]
+
+        try:
+            res = subprocess.Popen(
+                args,
+                stdin=subprocess.DEVNULL,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT
+            )
+            res.wait()
+            for line in res.stdout.read().decode().splitlines():
+                m = re.search(r'#define ([\w()]+) (.+)', line)
+                if m is not None:
+                    result.append("-D{}={}".format(m.group(1), m.group(2)))
+                else:
+                    m = re.search(r'#define (\w+)', line)
+                    if m is not None:
+                        result.append("-D{}".format(m.group(1)))
+        except FileNotFoundError:
+            self._warn("Cannot find compiler %s in PATH." % compiler)
+
+        return self._filter_defines(result)
+
+    def _get_default_include_paths(self, compiler, std, language):
+        import subprocess
+        import re
+
+        result = list()
+
+        args = [compiler, '-x', language]
+        if std is not None:
+            args += ['-std=' + std]
+        args += ['-Wp,-v', '-E', '-']
+
+        try:
+            res = subprocess.Popen(
+                args,
+                stdin=subprocess.DEVNULL,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT
+            )
+            res.wait()
+            pick = False
+            for line in res.stdout.read().decode().splitlines():
+                if '#include <...> search starts here:' in line:
+                    pick = True
+                    continue
+                if '#include "..." search starts here:' in line:
+                    pick = True
+                    continue
+                if 'End of search list.' in line:
+                    break
+                if pick:
+                    m = re.search(r'\s*(.*)$', line)
+                    if m is not None:
+                        result.append("-I{}".format(m.group(1)))
+        except FileNotFoundError:
+            self._warn("Cannot find compiler %s in PATH." % compiler)
+
+        return result
+
+    def _filter_defines(self, defines):
+        # Remove some default flags which get set by clang itself.
+        # Otherwise, we will get error later:
+        for flag in CompilerBuiltIns.__DEFINES_BLACKLIST:
+            prefix = "-D%s" % flag
+            for define in defines:
+                if define.startswith(prefix):
+                    defines.remove(define)
+        return defines
+
+    def _debug(self, msg):
```

Comment:
There seems to be no need for these functions from my point of view. The name of the class and function calling the log is already part of logging module.

Reply:
Okay, I just had in mind that in other files you used to have dedicated prefixes in the log messages, so I wanted to keep that aligned. If this is not required, I will happily convert to plain logging ;-)